yarn-project/aztec.js/src/wallet/account_wallet.ts

```diff
@@ -80,11 +81,33 @@ export class AccountWallet extends BaseWallet {
     authorized: boolean,
   ): ContractFunctionInteraction {
     const message = this.getMessageHash(messageHashOrIntent);
-    if (authorized) {
-      return new ContractFunctionInteraction(this, this.getAddress(), this.getApprovePublicAuthwitAbi(), [message]);
-    } else {
-      return this.cancelAuthWit(message);
-    }
+    return new ContractFunctionInteraction(this, AuthRegistryAddress, this.getSetAuthorizedAbi(), [
+      message,
+      authorized,
+    ]);
+  }
+
+  /**
+   * Returns a function interaction to cancel a message hash as authorized or revoked.
+   * @param messageHashOrIntent - The message or the caller and action to revoke
+   * @returns - A function interaction.
+   */
+  public cancelPublicAuthWit(
```

Comment:
I'm open to nuke it, was mainly to have a familiar api. Nevertheless, with the work I am doing on the static private authwits, we cannot do a `cancel` for that nicely as this anyway, so might just nuke both of the `cancel` in there since so they vanish at the same time?